pkg/pillar/zedcloud/deferred.go

```diff
@@ -24,97 +27,149 @@ import (
 // or AddDeferred to build a queue for each key
 
 type deferredItem struct {
-	buf           *bytes.Buffer
-	size          int64
-	url           string
-	zedcloudCtx   *ZedCloudContext
-	bailOnHTTPErr bool // Return 4xx and 5xx without trying other interfaces
-}
-
-type deferredItemList struct {
-	list []deferredItem
+	itemType         interface{}
+	key              string
+	lastModifiedTime time.Time
```

Comment:
But I don't think we should move the items which are updated to the end.
Instead use createdTime to sort. And in that case a sort isn't needed.

Reply:
As discussed internally we do not want to edit this time with new items with the same key coming. Seems we do not need this at all, removed.